api/index.js

```diff
@@ -36,6 +36,8 @@ const handler = async (req, res, client, domain) => {
     await getSummaryOfEligibleDealsForAllocator(req, res, client, segs[1])
   } else if (segs[0] === 'inspect-request' && req.method === 'GET') {
     await inspectRequest(req, res)
+  } else if (segs[0] === 'ingest-eligible-deals' && req.method === 'POST') {
```

Comment:
Discussion point: what's a good endpoint name?

What I have right now:

```
POST /ingest-eligible-deals
```

Alternatives:
- `POST /eligible-deals` (But: such endpoint often expects only one deal, not an array of deals.)

Reply:
Good reasoning, I would also expect `POST /eligible-deals` to expect one object. Either we make the POSTer perform multiple requests, or I suggest we rename to something like `POST /eligible-deals-bulk`? It's unexpected to have a verb in there

Reply:
> Have we thought of using something like express.js for the routing, so we do not have that much if else logic?

We're rolling out Fastify atm, see other PRs :)

Reply:
> Also, is there a difference between eligible deals and active deals? Just for consistent naming.

An eligible deal is a deal that Spark considers eligible for retrieval testing.

An active deal is any deal that has not expired yet or was not slashed.

On second thought, the name "active deals" I used in spark-observer may not be the most precise one since spark-observer is tracking active FIL+ deals only. 🤔

Reply:
> Are we in risk of reaching the ingress limit here? I think it is usually like 2 or 5 MB.

Great question! What do you mean by the ingress limit? Do you mean the maximum size of the request body allowed by Cloudflare reverse proxy? I don't know what's the limit.

As I see it, it's the responsibility of the client sending there requests (the code in deal-observer) to batch new deals in requests that are small enough to fit into the limit.

In the pull request, the server rejects request with a body larger than 100MB. This can be easily tweaked.

```
const body = await getRawBody(req, { limit: '100mb' })
```

Reply:
Re: endpoint name - I'll use `POST /eligible-deals-batch`

Reply:
Please don't forget to update the PR name to match